llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7652,6 +7652,8 @@ DenseMap<const SCEV *, Value *> LoopVectorizationPlanner::executePlan(
   VPlanTransforms::unrollByUF(BestVPlan, BestUF,
                               OrigLoop->getHeader()->getContext());
   VPlanTransforms::optimizeForVFAndUF(BestVPlan, BestVF, BestUF, PSE);
+  VPlanTransforms::simplifyRecipes(BestVPlan, Legal->getWidestInductionType());
```

Comment:
It looks like the code to refactor simplifyRecipes could actually be its own NFC patch, right?

Reply:
Do you mean just exposing it in VPlanTransforms.h without adding new users outside of VPlanTransforms.cpp?

Reply:
I just meant the changes to pull out the `removeDeadRecipes` call from `simplifyRecipes ` and call `simplifyRecipes` and `removeDeadRecipes` here from `executePlan`. It looked like these changes are notionally independent from your changes in `simplifyRecipe` for `VPScalarIVStepsRecipe`